src/pages/EnvironmentsPage.ts

```diff
@@ -1,14 +1,40 @@
+import { NavLinkProvider } from '@terra-ui-packages/core-utils';
 import { h } from 'react-hyperscript-helpers';
-import { Environments } from 'src/analysis/Environments/Environments';
+import { EnvironmentNavActions, Environments } from 'src/analysis/Environments/Environments';
 import FooterWrapper from 'src/components/FooterWrapper';
 import TopBar from 'src/components/TopBar';
-import * as Nav from 'src/libs/nav';
+import { terraNavKey, TerraNavLinkProvider, terraNavLinkProvider } from 'src/libs/nav';
+
+type NavMap<NavTypes, FnReturn> = {
+  [Property in keyof NavTypes]: (args: NavTypes[Property]) => FnReturn;
+};
+
+export const makeNavProvider = (terraNav: TerraNavLinkProvider): NavLinkProvider<EnvironmentNavActions> => {
```

Comment:
Not necessarily needed now, but it would be nice to have a helper to construct these objects. It looks like, in most cases, these nav providers could be constructed from a map of component route names to Terra UI route names (`workspace-view` => `workspace-dashboard` in this case).

Reply:
agreed,  I saw the same eventuality.  But I left it as a potential TODO once n is 2 or 3.